packages/eslint-plugin/src/rules/css-lint/css-lint.ts

```diff
@@ -0,0 +1,180 @@
+  name: 'css-lint',
+  meta: {
+    type: 'problem',
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          generateMetadata: {
+            type: 'boolean',
+          },
+          modules: {
+            description: 'Configures the import name and source of Griffel',
+            type: 'array',
+            items: {
+              type: 'object',
+              required: ['moduleSource', 'importName'],
+              properties: {
+                moduleSource: {
+                  type: 'string',
+                },
+                importName: {
+                  type: 'string',
+                },
+              },
+            },
+          },
+          babelOptions: {
+            description:
+              'Configures the Griffel babel preset, see https://github.com/microsoft/griffel/blob/main/packages/babel-preset/README.md',
+            type: 'object',
+            properties: {
+              plugins: {
+                type: 'array',
+              },
+              presets: {
+                type: 'array',
+              },
+            },
+          },
+          evaluationRules: {
+            description:
+              'Configures the Griffel babel preset, see https://github.com/microsoft/griffel/blob/main/packages/babel-preset/README.md',
+            type: 'array',
+            items: {
+              type: 'object',
+              required: ['action'],
+              properties: {},
+            },
+          },
+          projectRoot: {
+            type: 'string',
+          },
+        },
+      },
+    ],
+    messages: {
+      foo: 'WIP - not ready yet',
+    },
+    docs: {
+      description: 'WIP - not ready yet',
+      recommended: 'error',
+    },
+  },
+  defaultOptions: [
+    {
+      generateMetadata: true,
+      modules: [
+        {
+          moduleSource: '@griffel/react',
+          importName: 'makeStyles',
+          resetImportName: 'makeResetStyles',
+        },
+        {
+          moduleSource: '@fluentui/react-components',
+          importName: 'makeStyles',
+          resetImportName: 'makeResetStyles',
+        },
+      ],
+    },
+  ],
+  create(context) {
+    // Map of all style slots to the makeStyles return function name
+    const slotPropertiesByDeclaratorId = new Map<string, Map<string, TSESTree.Property>>();
+    // map of all styles to the return of the makeResetStyles function name
+    const resetEntriesByDeclaratorId = new Map<string, TSESTree.CallExpression>();
+    const options = context.options[0];
+
+    return {
+      'Program:exit'(node) {
+        if (!context.getPhysicalFilename) {
+          throw new Error('No physical filename could be found');
+        }
+
+        const filename = context.getPhysicalFilename();
+        const source = context.getSourceCode().getText();
+        const { metadata } = transformSync(source, {
+          filename,
+          pluginOptions: {
+            generateMetadata: true,
+            ...options,
```

Comment:
```suggestion
            ...options,
            generateMetadata: true,
```

nit: You probably want to flip this to generate metadata always